_inc/client/state/settings/reducer.js

```diff
@@ -34,6 +35,11 @@ export const items = ( state = {}, action ) => {
 			return assign( {}, state, action.initialState.settings );
 		case JETPACK_SETTINGS_FETCH_RECEIVE:
 			return assign( {}, action.settings );
+		case JETPACK_MODULES_LIST_RECEIVE:
+			Object.keys( action.modules ).forEach( key => {
+				action.modules[ key ] = action.modules[ key ].activated;
+			} );
+			return assign( {}, action.modules );
```

Comment:
These four lines can be replaced with:
```js
return assign( {}, _.mapValues( action.modules, value => value.activated ) );
```
to avoid mutating `action`.

Reply:
@eliorivero the problem with updating all of the values is that it clears out the settings of things that aren't modules.  For example, it removes the PBE email address and all other sub-settings, which is why I was only updating the state for the module keys.  

![screen shot 2017-08-28 at 11 43 40 am](https://user-images.githubusercontent.com/7129409/29781130-2b858ba4-8be6-11e7-87da-b2be5ca61c17.png)